packs/actions/singin.ts

```diff
@@ -0,0 +1,44 @@
+import type { AppContext } from "$store/apps/site.ts";
+import { setCookies } from "$store/utils/cookies.ts";
+import { BAD_REQUEST, OK } from "$store/utils/enum.ts";
+
+export interface SingIn {
+  email: string;
+  password: string;
+}
+
+export default async function loader(
+  props: SingIn,
+  _req: Request,
+  ctx: AppContext,
+) {
+  const { email, password } = props;
+  const { supabaseClient } = ctx;
+
+  const passwordRegex = /^(?=.*[!@#$%^&*()_+{}:<>?])(?=.*[A-Z])(?=.*\d).{8,}$/;
+
+  if (!passwordRegex.test(password)) {
+    return {
+      status: BAD_REQUEST,
+      message:
+        "Sua senha tem que ter 8 caracteres, pelo menos um caracter especial, uma letra maiúscula e um número",
```

Comment:
Usa palavras chaves no backend para que no frontend vocês consigam criar formas mais flexiveis de exibir um erro ou mensagens personalizadas, imagine voce escrevendo um if tipo:

`if (data.message === "Sua senha tem que ter 8 caracteres, pelo menos um caracter especial, uma letra maiúscula e um número")`

```suggestion
        "password-strength",
```

Reply:
Estava pensando em usar essas mensagens de error no front de um vez.

Exemplo: <p>{response.message]</p>